spec/features/post_index_spec.rb

```diff
@@ -41,13 +41,10 @@
     visit user_posts_path(user1)
     expect(page).to have_content(post1.text)
   end
-  scenario 'Displays post title' do
-    visit user_posts_path(user1)
-    expect(page).to have_content(post1.title)
-  end
-  scenario 'Displays the 1st comment' do
-    visit user_posts_path(user1)
-    expect(page).to have_content(comment1.body)
+  scenario 'Displays the first comments on a post & total comments' do
+    visit user_posts_path(user)
+    expect(page).to have_content(comment2.text)
+    expect(page).to have_content("comments: #{post.comments_counter}")
```

Comment:
- You did well so far, but one of your tests is failing, please note all tests must pass to ensure your app works as expected without any bugs. Please fix the following issues 🙏 :
   - On line (45), you do not have a `user` variable in your test file, it should be `user1` 
   - On line (46). you do not have a `text` attribute in your comment table, it should be `body` instead
   - On line (47), you do not have a `post` variable in your test file, it should be `post1..post4`
![wrfwef](https://github.com/maths29/blog_app/assets/73607512/6eaf1b6d-5e8d-488c-b121-47fe57988a79)

- Regarding the requirements the previous reviewer suggested, about seeing the first comments on a post, I suggest you use the `recent_comments` method in your `post` model to ensure we can see the first comments on a post. Please fix this 🙏 
   ```ruby
       scenario 'Displays the first comments on a post & total comments' do
         visit user_posts_path(user1)
         user1.posts.each do |post| 
            post.recent_comments.each do |comment|
               expect(page).to have_content(comment.body)
            end
        expect(page).to have_content("comments: #{post.comments_counter}")
        end
     end
```

Reply:
**OPTIONAL** - Please note that for the above suggestions to work, you need to correct some issues in your `post` model. Your `recent_comments` method in the post model should look like this:
    ```ruby
        def recent_comments
            comments.order(created_at: :desc).limit(5)
        end
    ```